cdc/cdc.cc

```diff
@@ -103,17 +107,21 @@ static future<> setup_log(db_context ctx, const schema& s) {
     b.with_column("stream_id", uuid_type, column_kind::partition_key);
     b.with_column("time", timeuuid_type, column_kind::clustering_key);
     b.with_column("batch_seq_no", int32_type, column_kind::clustering_key);
-    b.with_column("operation", int32_type);
+    b.with_column("operation", data_type_for<operation_native_type>());
```

Comment:
What user needs to know the type? It will be an int-like type, describe schema will tell him. Yes, changing the underlying type of "operation" enum type would change this, but it is better that the definition is at _one_ place, i.e. in "enum class operation : int32_t" than spread out in several. 
Dependent types is a good thing.

Reply:
You're right that "describe schema" can tell a cqlsh user which type "operation" is, but I suspect that actual application code will not call "describe schema" but rather assume it has a specific type. An important part of the design document should have been (@haaawk do we have this?) the user's API to use this feature, and I think the type of these fields are part of the API, and we won't want to ever change it.
Anyway, I'm not **strongly** against this, just mildly, so if you prefer it this way, then so be it.

Reply:
Yes. But the docs will (or should) be generated by schema describe... > schema.txt; insert content of schema.txt into docs. 
Point being that, no, op type should not change (unless needed to), and as such is should be defined at one point, and that is the enum type. underlying_type_t<> is not something that changes willy-nilly, it is restricted by enum <name> : <type>. Again, I _really_ dislike spreading out loosly coupled definitions, so when I can make stuff dependent on a single definition, I do.